libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -247,31 +253,31 @@ class MapboxTripSession(
     }
 
     private fun updateRawLocation(rawLocation: Location) {
+        locationObservers.forEach { it.onRawLocationChanged(rawLocation) }
+        lastLocationUpdateTimeMillis = Date().time
```

Comment:
Date().time will become break if you change timezones right?

Reply:
As I know, `Date().time` returns count of milliseconds from 1 January 1970, so I think it is not depend on timezone. Wdyt?

Reply:
instead of using a `lastLocationUpdateTimeMillis` threshold

try using a `lastGetStatusTimeMillis` threshold

Reply:
but use elapsedRealitime 😉

Reply:
> As I know, Date().time returns count of milliseconds from 1 January 1970, so I think it is not depend on timezone. Wdyt?

Yeah but that time drifts and needs to be corrected. Use elapsedRealtime is since the device as been booted. It is guaranteed to always go forward, never backward (AKA monotonic)

Reply:
The get status call is a long operation, and it probably the source of collision issues.

location update loop
--o----o----------o---o

get status poll (s stands for start poller)
--s----s----o----os---s

result: S=S is a get status window. we can't have overlapping windows
--S=S-S=S-S=S-XXX--

^ X is the collision if you don't stall a get status call